lsp-mode.el

```diff
@@ -7423,6 +7118,9 @@ returns the command to execute."
   (when (functionp 'lsp-ui-mode)
     (lsp-ui-mode))
 
+  (when lsp-headerline-breadcrumb-enable
+    (add-hook 'lsp-managed-mode-hook 'lsp-headerline-breadcrumb-mode))
+
```

Comment:
Is this needed? The mode should have been enabled with `lsp-configure-hook` already

Reply:
Or you should add it into `lsp-configure-hook` instead of `lsp-managed-mode-hook`

Reply:
We need it because we need to auto-enable the breadcrumb mode if `lsp-headerline-breadcrumb-enable` is `t`.
This is actually not working and I don't know why yet, For some reason the `lsp` function does not run for me.

Reply:
`lsp` will run only once per buffer (unless you disable and reenable it).

Reply:
Also, I think you should really use `lsp-configure-hook` here instead.
Else this `lsp-headerline-breadcrumb-mode` will also be enabled again when `(lsp-managed-mode -1)`

Reply:
I see, I think it makes sense, I moved it to `lsp-configure-buffer`, I don't know if it is correct, but I added the hook and then later on the same function fire the lsp-configure-hook.

Now it works nice :)

Reply:
I think this part should stay in `lsp--auto-configure` function. Do you still have it not working even when toggle `lsp-mode` on/off again?

Reply:
In `lsp--auto-configure` I can't make trigger, that's was my previous problem.
now in `lsp-configure-buffer` it works even toggling `lsp-mode` on/off

Reply:
I mean when in `lsp--auto-configure`, even with `lsp-mode` off/on you still cannot making it triggered?
That seems weird.
Can you try to close the buffer, and open the buffer again?
It will trigger both `lsp` and `lsp-managed-mode`, which will add the hook and also trigger `lsp-configure-buffer`.

Reply:
Yeah, this keeps not working, even adding a `(message "-->")` [here](https://github.com/emacs-lsp/lsp-mode/blob/master/lsp-mode.el#L8420) and closing emacs and opening a lsp client does not print...
I wonder if it's something with my emacs, I use doom-emacs with [no-byte-compile enabled](https://github.com/ericdallo/dotfiles/blob/master/.doom.d/packages.el#L18)

Reply:
Forget it, I compiled lsp-mode and it seems to work, I'll try to move to `lsp--auto-configure` again

Reply:
@kiennq Well, after some many attempts, I could not reproduce it again, for some reason my `lsp` function is always compiled even using `no-byte-compile` in straight.el, or recompiling doom, but I agree that `lsp--auto-configure` is the right place so I moved to there.